drivers/media/i2c/adv7180.c

```diff
@@ -1266,6 +1267,18 @@ static int init_device(struct adv7180_state *state)
 			goto out_unlock;
 	}
 
+	/* Select first valid input */
+	for (i = 0; i < 31; i++) {
```

Comment:
Is it not possible to have 32 inputs? Is 30 the best maximum?

Reply:
The valid values are defined by ADV7180_INPUT_* and ADV7182_INPUT_* at line 156ish. They max out at 0x11.
Checking the datasheet, whilst INSEL is 5 bits, 10010 to 11111 are reserved, so there is no valid use case for it to be set to greater than 0x11.

Seeing as I need to update tc358743, I'll amend this to check the top bit anyway.